src/main/java/domain/board/PieceExtractor.java

```diff
@@ -0,0 +1,9 @@
+package domain.board;
+
+import domain.piece.Piece;
+import java.util.List;
+
+public interface PieceExtractor {
+
+    List<Piece> extract(List<BoardLocation> boardLocations);
+}
```

Comment:
함수형 인터페이스 사용을 의도하신 거라면 `@FunctionalInterface` 애노테이션 추가되면 좋을 것 같습니다!
```suggestion
@FunctionalInterface
public interface PieceExtractor {

    List<Piece> extract(List<BoardLocation> boardLocations);
}
```